include/copc-lib/io/writer.hpp

```diff
@@ -88,7 +92,13 @@ class Writer : public BaseIO
     // Adds a subpage to a given page
     Page AddSubPage(Page &page, VoxelKey key);
 
-  private:
+    virtual ~Writer()
+    {
+        if (writer_->IsOpen())
+            writer_->Close();
```

Comment:
I think we can change this to just call Close()?

Reply:
Actually, could we make ~Writer non-overriden and just have it call the overriden Close()? I *think* that should work?

```
~Writer() { this->Close(); }
virtual Close() { doClose() }

FileWriter:
Close() override {doClose2()}
// Don't override destructor
```

Reply:
I think it's bad practice to call virtual functions in destructor?